File: FrontEnd/src/app/features/reservations/customers/classes/dtos/customer-write-dto.ts
import { Guid } from 'guid-typescript'

export interface CustomerWriteDto {

    // PK
    id: number
    // FKs
    nationalityId: number
    taxOfficeId: Guid
    vatRegimeId: Guid
    // Fields
    description: string
    taxNo: string
    profession: string
    address: string
    phones: string
    personInCharge: string
    email: string
    balanceLimit: number
    isActive: boolean
    // Metadata
    putAt: string

}

File: API/Features/Reservations/Customers/Dtos/CustomerWriteDto.cs
using System;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations.Customers {

    public class CustomerWriteDto : IBaseEntity, IMetadata {

        // PK
        public int Id { get; set; }
        // FKs
        public Guid TaxOfficeId { get; set; }
        public Guid VatRegimeId { get; set; }
        public int NationalityId { get; set; }
        // Fields
        public string Description { get; set; }
        public string TaxNo { get; set; }
        public string Profession { get; set; }
        public string Address { get; set; }
        public string Phones { get; set; }
        public string PersonInCharge { get; set; }
        public string Email { get; set; }
        public decimal BalanceLimit { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}